File: LeetCode/Study Plan/Practice Problems/Hard/Orderly Queue/Cpp/orderly_queue.cc
#include <iostream>

std::string orderly_queue(std::string s, int k) {
    if (k == 1) {
        std::string ans = s;
        for (int i=0; i<s.length(); i++) {
            std::string temp = s.substr(i, s.length()) + s.substr(0, i);
            if (ans.compare(temp) > 0) {
                ans = temp;
            }
        }
        return ans;
    }else {
        sort(s.begin(), s.end());
        return s;
    }
}

int main() {
    std::string s = "cba";
    std::cout<<orderly_queue(s, 1)<<"\n";
}

File: LeetCode/Hard/Orderly Queue/Go/orderly_queue.go
package main

import (
	"fmt"
	"sort"
)

func orderly_queue(s string, k int) string {
	n := len(s)
	if k == 1 {
		ans := s
		for i := 0; i < n; i++ {
			temp := s[i:] + s[0:i]
			if temp < ans {
				ans = temp
			}
		}
		return ans
	} else {
		str := []byte(s)
		sort.Slice(str, func(i, j int) bool { return str[i] < str[j] })
		return string(str)
	}
}

func main() {
	fmt.Println(orderly_queue("cba", 1))
}
